src/main/java/com/imsweb/geocoder/GeocodeInput.java

```diff
@@ -187,6 +315,26 @@ public void setUseAliasTable(Boolean useAliasTable) {
         if (getUseAliasTable() != null)
             params.put("useAliasTable", getUseAliasTable() ? "true" : "false");
 
+        if (getShouldUseRelaxation() != null && !getShouldUseRelaxation())
+            params.put("r", "false");
+        else if (getRelaxedAttributes() != null) {
+            params.put("r", "true");
+            if (_RATTS_VALUES.containsAll(Arrays.asList(getRelaxedAttributes().split(","))))
+                params.put("ratts", getRelaxedAttributes());
```

Comment:
Do you think it's necessary to value the parameters? That would mean a new version is needed if they add a value. What does the geocoder do if it's given an invalid value? I'm not even sure an exception would be bad in this case, since we would be trying to use an option that doesn't exist.

Reply:
> What does the geocoder do if it's given an invalid value?

For the relaxed attributes and soundex attributes, invalid values appear to be ignored. For ref data source, an exception is indeed thrown.

Reply:
Your change looks good, but the build is failing now. Does the unit test need to be updated?

It also looks like the indentation is wrong, so please format the file.